coin_change_2.py

```diff
@@ -0,0 +1,31 @@
+#Tume complexity: O(n*T): where n: # of coins, T: target amount value
+#Space complexity: O(n*T)
+#This algorithm uses dynamic programming, by converting it into several subproblems, where each subproblem solves for
+# finding the number of ways to reach a target values, by first picking certain number of coins.
+# We keep on adding more number of coins, till we cover all coins, and target value is reached.
+
+import numpy as np
```

Comment:
I am not sure, but I don't think you'd be allowed to use external libraries. Please verify this with your instructor.